engine/verification/utils/fixtures.go

```diff
@@ -306,34 +341,103 @@ func LightExecutionResultFixture(chunkCount int) CompleteExecutionResult {
+// The chain is in the form of root <- R1,1 <- R1,2 <- ... <- C1 <- R2,1 <- R2,2 <- ... <- C2 <- ...
+// In this chain R refers to reference blocks that contain guarantees.
+// C refers to a container block that contains an execution receipt for its preceding reference blocks.
+// e.g., C1 contains an execution receipt for R1,1, R1,2, etc, and C2 contains a receipt for R2,1, R2,2, etc.
+// For sake of simplicity and test, container blocks (i.e., C) do not contain any guarantee.
 //
-// It returns a slice of CompleteExecutionResult fixtures that contains a pair of (Ri <- Ci).
-func CompleteExecutionResultChainFixture(t *testing.T, root *flow.Header, count int) []*CompleteExecutionResult {
-	results := make([]*CompleteExecutionResult, 0, count)
+// It returns a slice of complete execution receipt fixtures that contains a container block as well as all data to verify its contained receipts.
+func CompleteExecutionReceiptChainFixture(t *testing.T, root *flow.Header, count int, opts ...CompleteExecutionReceiptBuilderOpt) []*CompleteExecutionReceipt {
+	completeERs := make([]*CompleteExecutionReceipt, 0, count)
 	parent := root
+
+	builder := &CompleteExecutionReceiptBuilder{
+		resultsCount: 1,
+		copyCount:    1,
+		chunksCount:  1,
+		chain:        flow.Testnet.Chain(),
+	}
+
+	for _, apply := range opts {
+		apply(builder)
+	}
+
 	for i := 0; i < count; i++ {
 		// Generates two blocks as parent <- R <- C where R is a reference block containing guarantees,
 		// and C is a container block containing execution receipt for R.
-		result := CompleteExecutionResultFixture(t, 1, flow.Testnet.Chain(), parent)
-		results = append(results, &result)
+		allResults, allData, head := ExecutionResultsFromParentBlockFixture(t, parent, builder)
+		containerBlock := ContainerBlockFixture(head, allResults)
+		completeERs = append(completeERs, &CompleteExecutionReceipt{
+			ContainerBlock: containerBlock,
+			ReceiptsData:   allData,
+		})
+
+		parent = containerBlock.Header
+	}
+	return completeERs
+}
+
+// ExecutionResultsFromParentBlockFixture creates a chain of results from a parent block.
+//
+// By default each result refers to a distinct reference block, and it extends the block chain after generating each
+// result (i.e., for the next result).
+//
+// Each result may appear in more than one receipt depending on the builder parameters.
+func ExecutionResultsFromParentBlockFixture(t *testing.T, parent *flow.Header, builder *CompleteExecutionReceiptBuilder) ([]*flow.ExecutionResult,
+	[]*ExecutionReceiptData, *flow.Header) {
+	allData := make([]*ExecutionReceiptData, 0, builder.resultsCount)
+	allResults := make([]*flow.ExecutionResult, 0, builder.resultsCount)
+
+	for i := 0; i < builder.resultsCount; i++ {
+		result, data := ExecutionResultFromParentBlockFixture(t, parent, builder)
+
+		// makes several copies of the same result
+		for copy := 0; copy < builder.copyCount; copy++ {
```

Comment:
`copy` is a builtin function, does this compile?

Reply:
It complies since they have different types, however I refactored it in https://github.com/onflow/flow-go/pull/573/commits/58c7b3bf62d14714896ac44df80d5e6a437992e0.